spec/miscellaneous/lib/html_cleaner_spec.rb

```diff
@@ -584,7 +584,7 @@
 
     it "leaves audio tags alone" do
       original = "<audio controls=\"controls\" crossorigin=\"anonymous\" preload=\"metadata\">
-  <source src=\"http://example.com/podfic.mp3\" type=\"audio/mpeg\"></source>
+  <source src=\"http://example.com/podfic.mp3\" type=\"audio/mpeg\">
```

Comment:
This is where the test failure is. A `</source>` is getting added right before the `</audio>`.

Reply:
This is tricky. Sanitize 5 (and our MediaSanitizer, by extension) does not add closed tags for `<source>`, because it uses Nokogumbo 2.0's HTML5 serializer (`Nokogiri::HTML5.fragment`, rubys/nokogumbo#14). If you move this spec to `media_sanitizer_spec`, it will pass.

Our HTMLCleaner uses Nokogiri's old HTML serializer (`Nokogiri::HTML5.fragment`) which does add closed tags for `<source>`.

https://github.com/otwcode/otwarchive/blob/c364d7f1a76af0d73a260158bd824e1c52962637/lib/html_cleaner.rb#L172-L174

Other things using the old HTML serializer: the word counter, the importing story parser. That's a lot of stuff to update.

Reply:
Ugh, okay, here we go...

> If you move this spec to `media_sanitizer_spec`, it will pass. 

Which position reflects the _actual_ behavior we'd get? 

"Sarken, you should be able to answer that by trying it on your development environment," you say. Which is a good point, but when I tried the embeds on my dev environment, it continued to insert the `</source>` and `</track>` tags. Because @elzj had mentioned this resolved the issue, I assumed my dev environment was just being cranky and not updating, so I shrugged and wrote automated tests to prove it was fixed instead. 

Now that you've said this, though, I have the sinking feeling the behavior I was seeing is correct and indicates updating Sanitize does not by itself fix the unwanted tags.

Reply:
I think `html_cleaner_spec` is closer to the actual behavior. We don't use MediaSanitizer on its own, but as part of HtmlCleaner.

Reply:
Yeeeeah, I figured. idk about you, but even though I don't love releasing this with invalid HTML, I'm even less keen on shoving all of those changes into this release, especially considering this release has taken -- let me check my watch -- almost a month, and we've been struggling to find folks to test the changes to the sanitizer.